tests/performance_tests/log_reader.py

```diff
@@ -70,3 +78,51 @@ def scrapeLog(total, path):
                         total.updateTotal(int(value[2]), 0, 0, 0, 0, int(v2Logging[0]))
                 else:
                     print("Error: Unknown log format")
+
+def findPrunedRangeOfInterest(data: chainData, numAddlBlocksToDrop=0):
+
+    startBlockIndex = 0
+    endBlockIndex = len(data.blockLog) - 1
+
+    #skip leading empty blocks in initial range of interest as well as specified number of potentially non-empty blocks
+    droppedBlocks = 0
+    for block in data.blockLog:
+        if block.blockNum < data.startBlock or (droppedBlocks == 0 and block.transactions == 0):
+            continue
+        else:
+            if droppedBlocks < numAddlBlocksToDrop:
+                droppedBlocks += 1
+                continue
+            else:
+                startBlockIndex = data.blockLog.index(block)
+                break
+
+    #skip trailing empty blocks at end of initial range of interest as well as specified number of potentially non-empty blocks
+    droppedBlocks = 0
+    for block in reversed(data.blockLog):
+        if block.blockNum > data.ceaseBlock or (droppedBlocks == 0 and block.transactions == 0):
+            continue
+        else:
+            if droppedBlocks < numAddlBlocksToDrop:
+                droppedBlocks += 1
+                continue
+            else:
+                endBlockIndex = data.blockLog.index(block)
+                break
+
+    return startBlockIndex,endBlockIndex
+
+def scoreTransfersPerSecond(data: chainData, numAddlBlocksToDrop=0) -> stats:
+    consecutiveBlockTps = []
+
+    startBlockIndex,endBlockIndex = findPrunedRangeOfInterest(data, numAddlBlocksToDrop)
+
+    if startBlockIndex >= endBlockIndex:
+        print(f"Error: Invalid block index range start: {startBlockIndex} end: {endBlockIndex}")
+        return stats()
+
+    for i in range(startBlockIndex, endBlockIndex+1):
+        if i + 1 < endBlockIndex:
+            consecutiveBlockTps.append(data.blockLog[i].transactions + data.blockLog[i+1].transactions)
+
```

Comment:
I think you'd prefer to do a single conversion from Python list to numpy array, so rather than passing `consecutiveBlockTps` to each numpy function: `npTps = numpy.array(consecutiveBlockTps, dtype=numpy.uint32)` and pass `npTps` to each function.  I don't think the interpreter can be depended upon to make that optimization for you.

Reply:
Good point. I'll update.